cpp/src/arrow/compare.cc

```diff
@@ -523,24 +524,23 @@ class ArrayEqualsVisitor : public RangeEqualsVisitor {
 
 template <typename TYPE>
 inline bool FloatingApproxEquals(const NumericArray<TYPE>& left,
-                                 const NumericArray<TYPE>& right) {
+                                 const NumericArray<TYPE>& right,
+                                 typename TYPE::c_type epsilon) {
   using T = typename TYPE::c_type;
 
   const T* left_data = left.raw_values();
   const T* right_data = right.raw_values();
 
-  static constexpr T EPSILON = static_cast<T>(1E-5);
-
   if (left.null_count() > 0) {
     for (int64_t i = 0; i < left.length(); ++i) {
       if (left.IsNull(i)) continue;
```

Comment:
I know this is not related to this commit, but should it check right.IsNull?

Reply:
No, null bitmaps are compared in [BaseDataEquals](https://github.com/apache/arrow/blob/df562103ab9a17540b0dd8fe01842e94a78fff31/cpp/src/arrow/compare.cc#L583-L586)